packages/beacon-node/src/chain/seenCache/seenGossipBlockInput.ts

```diff
@@ -443,16 +441,26 @@ export class SeenGossipBlockInput {
     this.emitter.emit(ChainEvent.publishDataColumns, sampledColumns);
 
     for (const column of sampledColumns) {
-      this.getGossipBlockInput(
-        config,
-        {
-          type: GossipedInputType.dataColumn,
-          dataColumnSidecar: column,
-          dataColumnBytes: null,
-        },
-        // TODO: Pass in metrics. Should this use a different availability source?
-        null
+      blockCache.cachedData.dataColumnsCache.set(column.index, {dataColumn: column, dataColumnBytes: null});
+    }
+
+    if (blockCache.block) {
```

Comment:
Lets move this `if` down below the `blockCache.cachedData.resolveAvailability(blockData)` so that always gets called if we reconstruct the columns.  And only `resolveBlockInput` if we have the block.   I was a bit confused and looked at the code again.  There are two promises that get resolved.  One for data and one for the block and those promises are `await`ed in different locations/contexts

Reply:
I think resolveAvailability only gets called in the original function if we also have the block though, unless I'm mistaken? It doesn't look like any promises get resolved unless `signedBlock !== undefined`